sky/jobs/controller.py

```diff
@@ -291,13 +313,29 @@ def _run_one_task(self, task_id: int, task: 'sky.Task') -> bool:
                     continue
                 elif job_status in job_lib.JobStatus.user_code_failure_states():
                     # The user code has probably crashed, fail immediately.
-                    end_time = managed_job_utils.get_job_timestamp(
-                        self._backend, cluster_name, get_end_time=True)
+                    try:
+                        end_time = managed_job_utils.get_job_timestamp(
+                            self._backend, cluster_name, get_end_time=True)
```

Comment:
Should we have the function to handle the exceptions inside?

Reply:
That seems counterintuitive, could break some other use of the function that would not expect this fallback behavior.